lib/services/DatabaseService.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:developer';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class DatabaseService {
+  final databaseInstance = Firestore.instance.collection("MAC_Address");
+
+  // Sets up class to be singleton
+  DatabaseService._privateConstructor();
+  static final DatabaseService _instance = DatabaseService._privateConstructor();
+  factory DatabaseService() {
+    return _instance;
+  }
+
+  // Queries Firebase CloudFirestore based on the provided mac address
+  // Returns a string representing the vendor associated with the given address
+  // Returns a 'null' if the mac address is in the DB but with no vendor
+  // Throws 'NoSuchMethodError' if mac address is not in DB
```

Comment:
Maybe instead of throwing an error when the MAC address is not found in the database, it can return `null` and add the MAC address to the database.